server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java

```diff
@@ -554,6 +580,11 @@ public SystemIndexDescriptor getDescriptorCompatibleWith(Version version) {
         return null;
     }
 
+    public static int getDescriptorHash(String mappings, Settings settings) {
+        Map<String, Object> convertedMappings = XContentHelper.convertToMap(XContentType.JSON.xContent(), mappings, false);
```

Comment:
Should the final argument here be `true` to return an ordered map?

I can imagine that for a given version of Java a hash map with particular values inserted in a particular order will have the same structure. But that's not guaranteed across different versions of Java is it? So to ensure the same hash value on the line below it seems best to explicitly ask for an ordered map.